tests/test_pfcwd.py

```diff
@@ -0,0 +1,98 @@
+            dvs.runcmd("counterpoll queue enable")
+            time.sleep(10)
+
+            config_qualifiers = {
+                "TC" : PFCWD_TC[0],
+                "IN_PORTS": "Ethernet0"
+            }
+
+            expected_sai_qualifiers = {
+                "SAI_ACL_ENTRY_ATTR_FIELD_TC" : dvs_acl.get_simple_qualifier_comparator("3&mask:0xff"),
+                "SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS": dvs_acl.get_port_list_comparator(["Ethernet0"])
+            }
+        
+            dvs_acl.create_acl_rule(PFCWD_TABLE_NAME, PFCWD_RULE_NAME_1, config_qualifiers, action="DROP")
+            time.sleep(5)
+            rule_oid = self.find_acl_rule_oid(dvs)
+            dvs_acl.verify_acl_rule(expected_sai_qualifiers, action="DROP", acl_rule_id=rule_oid)
+
+            config_qualifiers = {
+                "TC" : PFCWD_TC[0],
+                "IN_PORTS": "Ethernet0,Ethernet16"
+            }
+
+            expected_sai_qualifiers = {
+                "SAI_ACL_ENTRY_ATTR_FIELD_TC" : dvs_acl.get_simple_qualifier_comparator("3&mask:0xff"),
+                "SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS": dvs_acl.get_port_list_comparator(["Ethernet0","Ethernet16"])
+            }
+
+            dvs_acl.update_acl_rule(PFCWD_TABLE_NAME, PFCWD_RULE_NAME_1, config_qualifiers, action="DROP")
+            time.sleep(5)
+            rule_oid = self.find_acl_rule_oid(dvs)
+            dvs_acl.verify_acl_rule(expected_sai_qualifiers, action="DROP",acl_rule_id=rule_oid)
+            dvs_acl.remove_acl_rule(PFCWD_TABLE_NAME, PFCWD_RULE_NAME_1)
+
+            config_qualifiers = {
+                "TC" : PFCWD_TC[1],
+                "IN_PORTS": "Ethernet8"
+            }
+
+            expected_sai_qualifiers = {
+                "SAI_ACL_ENTRY_ATTR_FIELD_TC" : dvs_acl.get_simple_qualifier_comparator("4&mask:0xff"),
+                "SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS": dvs_acl.get_port_list_comparator(["Ethernet8"]),
+            }
+
+            dvs_acl.create_acl_rule(PFCWD_TABLE_NAME, PFCWD_RULE_NAME_2, config_qualifiers, action="DROP")
+            time.sleep(5)
+            rule_oid = self.find_acl_rule_oid(dvs)
+            dvs_acl.verify_acl_rule(expected_sai_qualifiers, action="DROP",acl_rule_id=rule_oid)
+
+            config_qualifiers = {
+                "TC" : PFCWD_TC[1],
+                "IN_PORTS": "Ethernet8,Ethernet24"
+            }
+
+            expected_sai_qualifiers = {
+                "SAI_ACL_ENTRY_ATTR_FIELD_TC" : dvs_acl.get_simple_qualifier_comparator("4&mask:0xff"),
+                "SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS": dvs_acl.get_port_list_comparator(["Ethernet8","Ethernet24"]),
+            }
+
+            dvs_acl.update_acl_rule(PFCWD_TABLE_NAME, PFCWD_RULE_NAME_2, config_qualifiers, action="DROP")
+            time.sleep(5)
+            rule_oid = self.find_acl_rule_oid(dvs)
+            dvs_acl.verify_acl_rule(expected_sai_qualifiers, action="DROP",acl_rule_id=rule_oid)
```

Comment:
IIRC if you leave off the OID it will automatically run a similar method to what you wrote in `find_acl_rule_oid`, so I think you can remove that method and just run this:

```suggestion
            dvs_acl.verify_acl_rule(expected_sai_qualifiers, action="DROP")
```